articles/howtos/create-new-decorator.md

```diff
@@ -0,0 +1,222 @@
+---
+uid: CreateNewDecorator
+---
+# How can I create a new Decorator?
+
+Click [here](xref:Decorator) to find out more about Decorators.
+
+This how-to will serve as an example of how to create a new Decorator that will inject code at runtime for a Template that has been setup for it.
+
+This how-to will build up from the [previous](xref:MakeTemplateDecoratorReady) how-to, which showed how to make a Template ready for Decorators. Please interpolate on your end what is written here to make the necessary adjustments on what you're trying to achieve.
+
+## Create a new Decorator
+
+Using the the Intent Architect Application's Module Builder, it's easy to scaffold and wire up a new decorator.
+
+![Startup](images/create-new-decorator/MyDecoratorModuleStartup.png)
+
+Go ahead and create your Application in Intent Architect that will be used to create the Module that will contain the new Decorator for the `StartupTemplate`.
+
+Once the application creation is complete, right-click on the `Module Builder` node to create a new Decorator called `StaticFileServerDecorator`.
+
+![New Decroator](images/create-new-decorator/NewDecorator.png)
+
+We need to specify the Contract name for that Decorator so that it will know for which Template it is meant to Decorate.
+
+To recap, the Template we're trying to Decorate is the `StartupTemplate` and it requires a Decorator that adheres to this interface:
+
+```csharp
+public interface IStartupTemplateContract : ITemplateDecorator
+{
+    string ConfigureCode();
+}
+```
+
+It has a Full Namespace of: `MyCompany.MyModule.Templates.StartupTemplate.IStartupTemplateContract`
+
+So that is the value that we need to specify in the `Full TypeName` field for that Decorator.
+
+Ensure the `Decorator Settings`'s `Declare Usings` box is checked.
+
+![Add Type FullName](images/create-new-decorator/AddFQDN.png)
+
+Run the software factory.
+
+![Code Generation Staging](images/create-new-decorator/CodeGenerationStaging.png)
+
+Open Visual Studio and navigate to where the code has been generated and open up the `MyCompany.MyDecoratorModule` solution.
+
+You will notice that it has generated a Decorator as well as a Registration for the Decorator.
```

Comment:
I was originally thinking it's a bit superfluous, but I'm thinking just a minor change in wording will avoid that.

```suggestion
Observe that `StaticFileServerDecorator.cs` and `StaticFileServerDecoratorRegistration.cs` were generated.
```